### PR TITLE
Feature/add roast

### PR DESCRIPTION
**Pull request by florianblaschke**
With this i created a basic layout for navigation and implemented the Create-Operation for new Roasts as well as two tests. As with the last PR, lots of styling.

### CODE REVIEW

File: components/Form/index.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import { styled } from "styled-components";
```
Comment (Kheirah) [inline components/Form/index.js:1]: I wonder, isn't `styled` a **default export?** I was under the impression it's not a named one.

File: components/Layout/navigation.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import Link from "next/link";
+import { styled } from "styled-components";
```
Comment (Kheirah) [inline components/Layout/navigation.js:2]: Same goes here, `styled` might be a default export.

File: pages/api/roasts/index.js [hunk 3]
```diff
@@ -15,4 +15,21 @@ export default async function handler(req, res) {
...
+
+    if (existingRoast) {
+      return res.status(400).json({ error: "Roast already exists" });
+    }
+
+    try {
+      await Roast.create(newRoast);
+      return res.status(201).json({ message: "New roast created" });
+    } catch (error) {
+      return res.status(418).json({ error: error.message });
```
Comment (Kheirah) [inline pages/api/roasts/index.js:32]: **I'm a teapot!**

**Review by Kheirah — APPROVED** [review]
Very good!

Check the import of `styled-components`. It could be a default one. If it is, it could also be the reason for the hydration error. Maybe it'll resolve when you import `styled` as default. Try it out!

File: pages/api/roasts/index.js (hunk 3 above)
Comment (florianblaschke) [reply pages/api/roasts/index.js:32]: I found this one and just had to implement it!

File: components/Layout/navigation.js (hunk 2 above)
Comment (florianblaschke) [reply components/Layout/navigation.js:2]: Works both ways somehow. I changed it! Hydration Error still persists though ...

**Pull request merged**